src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -154,9 +155,9 @@ public AsymmetricSignatureProvider(SecurityKey key, string algorithm, bool willC
                 throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10634, (algorithm ?? "null"), key)));
 
             WillCreateSignatures = willCreateSignatures;
-            ValidateAsymmetricSecurityKeySize(key, algorithm, WillCreateSignatures);
+            _keySizeIsValid = new Lazy<bool>(ValidKeySize);
```

Comment:
@brentschmaltz That is a good question. We don't need the lazy, I included it to keep `ValidateAsymmetricSecurityKeySize` as only being called once.  Previously in the ctor, now once in the Lazy.

It would simplify this if we move the call to CreateAsymmetricAdapter. Do you think it is important to keep ValidateAsymmetricSecurityKeySize as only being called once?